ZionEscape/Scene.h

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include "SceneSpawner.h"
+
+using namespace System::Drawing;
+using namespace System::Collections::Generic;
+
+ref class Scene {
+  Bitmap^ image;
+  bool up, down, right, left;
+  List<SceneSpawner^>^ spawners;
+  Rectangle drawingArea;
+public:
+  Scene(bool up, bool down, bool right, bool left, Point pos) {
+    this->spawners = gcnew List<SceneSpawner^>;
+    this->up = up;
+    this->down = down;
+    this->right = right;
+    this->left = left;
+    this->ImageSelector();
+    this->drawingArea = Rectangle(pos.X, pos.Y, this->image->Width, this->image->Height);
+  }
+
+  ~Scene() {
+    for (short i = 0; i < this->spawners->Count; i++)
+      delete this->spawners[i];
```

Comment:
Mejor si lo haces con un `for each`.

```suggestion
    for each (SceneSpawner^ spawner in this->spawners)
      delete spawner;
```